torch/export/_trace.py

```diff
@@ -1472,7 +1480,93 @@ def wrapped_fn(*args):
                     non_strict_root, assigned_buffers
                 )
 
-            with ctx:
+            def custom_getattribute(self, attr, *, original_getattr, attrs_to_proxy):
+                """
+                The idea here is that we override subclass getattr methods to proxy
+                inner tensors and metadata. Because of infinite loop shenanigans, we have
+                to manually construct the getattr proxy nodes without relying on torch function
+                system.
+                """
+                out = original_getattr(self, attr)
+                if attr in attrs_to_proxy:
+                    if torch._C._is_torch_function_mode_enabled():
+                        # When we get here there is no guarantee that we will hit the
+                        # PreDispatchTorchFunctionMode, so we manually peak into the torch
+                        # function mode list and tweak the PreDispatchTorchFunctionMode.
+                        # This has side effect of proxying stuff like
+                        # proxy.node.meta["val"] = extract_val(val) because at that time, torch function
+                        # mode is still active. It seems bad to turn it off inside proxy_tensor.py, so
+                        # I guess we will just rely on DCE for now to remove extra stuff like detach
+                        torch_function_mode_stack = (
+                            torch.overrides._get_current_function_mode_stack()
+                        )
+                        for mode in torch_function_mode_stack:
+                            if isinstance(mode, PreDispatchTorchFunctionMode):
+                                tracer = mode.tracer
+                                proxy = get_proxy_slot(self, tracer).proxy
+                                inner_proxy = tracer.create_proxy(
+                                    "call_function", builtins.getattr, (proxy, attr), {}
+                                )
+                                # Even though the name says tensor tracking, the impl
+                                # suggests it tracks everything?
```

Comment:
I think in practice it tracks tensors, and also SymInts (once SymInt support was added after-the-fact)